hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -597,27 +596,40 @@ protected boolean checkDirectoryAlreadyExists(String volumeName,
   }
 
   /**
-   * Return volume info for the specified volume. If the volume does not
-   * exist, returns {@code null}.
+   * Return volume info that updated usageBytes for the specified volume.
    * @param omMetadataManager
    * @param volume
+   * @param updateUsage
    * @return OmVolumeArgs
    * @throws IOException
    */
-  protected OmVolumeArgs getVolumeInfo(OMMetadataManager omMetadataManager,
-      String volume) {
-
-    OmVolumeArgs volumeArgs = null;
-
-    CacheValue<OmVolumeArgs> value =
-        omMetadataManager.getVolumeTable().getCacheValue(
-        new CacheKey<>(omMetadataManager.getVolumeKey(volume)));
-
-    if (value != null) {
-      volumeArgs = value.getCacheValue();
-    }
+  protected static synchronized OmVolumeArgs syncUpdateUsage(
+      OMMetadataManager omMetadataManager, String volume, long updateUsage) {
+    OmVolumeArgs volumeArgs = omMetadataManager.getVolumeTable().getCacheValue(
+        new CacheKey<>(omMetadataManager.getVolumeKey(volume)))
+        .getCacheValue();
+    volumeArgs.getUsedBytes().add(updateUsage);
+    return volumeArgs.copyObject();
+  }
 
-    return volumeArgs;
+  /**
+   * Return volume info that updated usageBytes for the specified volume. And
+   * check Volume usageBytes quota.
+   * @param omMetadataManager
+   * @param volume
+   * @param updateUsage
+   * @return OmVolumeArgs
+   * @throws IOException
+   */
+  protected static synchronized OmVolumeArgs syncCheckAndUpdateUsage(
```

Comment:
I don't prefer to add new methods here, this makes current PR not clear to understand.
@captainzmc , can you make the minor adjustment for getVolumeInfo as I suggested in JIRA HDDS-4308.

Reply:
Thanks for @linyiqun's review.
Modifying getVolumeInfo to synchronized and get copyObject would not suffice. There is only one instance of volumeArgs in memory, and we need to update volumeArgs atomic after getVolumeInfo. Then get the value of the copyObject.
So, I made a modification based on your suggestion, added update volumeArgs's usedBytes to the getVolumeInfo method.

Reply:
I have not understood one part, how without volume lock this will help here.  

Because other threads can be updating volumeArgs when this update is happening/ other threads read omVolumeArgs.

Reply:
Do we need LongAdder still?

Reply:
> I have not understood one part, how without volume lock this will help here.
> 
> Because other threads can be updating volumeArgs when this update is happening/ other threads read omVolumeArgs.

This is a possible case. Can lead to update usageBytes inaccurate. 
Suppose that ThreadA update VolumeArgs using OMVolumeSetOwnerRequest, it gets VolumeArgs from DB if usageBytes is 1000 at this point. When ThreadB writes the new key using OMKeyCreateRequest, he changes usageBytes to 1100.
There is no Volume lock at this point, so ThreadB may finish executing first and ThreadA last. The Final usageBytes for VolumeArgs would be 1000. This should be a problem.
Maybe we still can't avoid using Volume lock. Any suggestions here? @linyiqun

Reply:
> Do we need LongAdder still?
LongAdder will no longer be necessary if there are no concurrent updates to usedBytes.

Reply:
> Do we need LongAdder still?

LongAdder will no longer be necessary if there are no concurrent updates to usedBytes.

Reply:
>Thank for @bharatviswa504's feedback. This is a possible case. Can lead to update usageBytes inaccurate.
Suppose that ThreadA update VolumeArgs using OMVolumeSetOwnerRequest, it gets VolumeArgs from DB if usageBytes is 1000 at this point. When ThreadB writes the new key using OMKeyCreateRequest, he changes usageBytes to 1100.

Good catch, @captainzmc . Seems we still use volume lock to help us do the volume quota update control. We should acquire volume lock before get volume info (copy volume info) and release volume lock after add the OMClientResponse into double buffer cache.